test/typescript/styles.spec.tsx

```diff
@@ -120,18 +121,19 @@ function OverridesTheme() {
 }
 
 // withTheme
-const ThemedComponent: React.SFC<{ theme: Theme }> = ({ theme }) => (
+const ThemedComponent: React.SFC<WithTheme> = ({ theme }) => (
   <div>{theme.spacing.unit}</div>
 );
 const ComponentWithTheme = withTheme()(ThemedComponent);
 
 // withStyles + withTheme
-interface AllTheProps {
-  theme: Theme;
+interface AllTheProps extends WithTheme {
   classes: { root: string };
 }
 
-const AllTheComposition = withTheme()(withStyles(styles)(({ theme, classes }) => (
+const AllTheComposition = withTheme()(
+  withStyles(styles)(
+    ({ theme, classes }: AllTheProps) => (
```

Comment:
This right here is why I had the `P` on the first parameter group... even though this change is arguably more correct, it causes a breakdown in type inference for some reason which requires you to add annotations like this. Are we sure we want to increase the type annotation burden like that? I'm not sure what it buys us...

Reply:
Because when you use it the way it is, it complains that theme is not provided.

```
const X = withTheme()((props) => <div/>))

<X /> // error theme not provided
```

Reply:
Hmm, that's odd, it's compiling fine for me...

Reply:
I'm adding a few more uses of the elements to confirm for the future.  We were defining but not using.